packages/agent/src/agent/http/index.ts

```diff
@@ -464,44 +466,78 @@ export class HttpAgent implements Agent {
     transformedRequest = await id.transformRequest(transformedRequest);
 
     const body = cbor.encode(transformedRequest.body);
-
-    this.log.print(
-      `fetching "/api/v2/canister/${ecid.toText()}/call" with request:`,
-      transformedRequest,
-    );
-
-    // Run both in parallel. The fetch is quite expensive, so we have plenty of time to
-    // calculate the requestId locally.
     const backoff = this.#backoffStrategy();
-    const request = this.#requestAndRetry({
-      request: () =>
+    try {
+      // Attempt v3 sync call
+      const requestSync = () =>
+        this.#fetch('' + new URL(`/api/v3/canister/${ecid.toText()}/call`, this.host), {
+          ...this.#callOptions,
+          ...transformedRequest.request,
+          body,
+        });
+
+      const requestAsync = () =>
         this.#fetch('' + new URL(`/api/v2/canister/${ecid.toText()}/call`, this.host), {
           ...this.#callOptions,
           ...transformedRequest.request,
           body,
-        }),
-      backoff,
-      tries: 0,
-    });
+        });
+
+      this.log.print(
+        `fetching "/api/v3/canister/${ecid.toText()}/call" with request:`,
```

Comment:
Looks like this log message is inaccurate if callSync is false

Reply:
Thanks - I've moved the log inside of each request function